File: isma-tools/src/main/java/ru/nstu/isma/generate/MemoryJavaFileObject.java
package ru.nstu.isma.generate;

import javax.tools.SimpleJavaFileObject;
import java.net.URI;

/**
 * Created by Bessonov Alex
 * on 10.12.2014.
 */
public class MemoryJavaFileObject extends SimpleJavaFileObject {

    private CharSequence content;

    public MemoryJavaFileObject(String className, CharSequence content) {
        super(URI.create("string:///" + className.replace('.', '/') + Kind.SOURCE.extension), Kind.SOURCE);
        this.content = content;
    }

    @Override
    public CharSequence getCharContent(boolean ignoreEncodingErrors) {
        return content;
    }

}

File: isma-compiler/jvm-utils/src/main/kotlin/ru/nstu/isma/generate/MemoryJavaFileObject.kt
package ru.nstu.isma.generate

import java.net.URI
import javax.tools.JavaFileObject
import javax.tools.SimpleJavaFileObject

/**
 * Created by Bessonov Alex
 * on 10.12.2014.
 */
class MemoryJavaFileObject(className: String, private val content: CharSequence?) : SimpleJavaFileObject(
    URI.create("string:///" + className.replace('.', '/') + JavaFileObject.Kind.SOURCE.extension),
    JavaFileObject.Kind.SOURCE
) {
    override fun getCharContent(ignoreEncodingErrors: Boolean): CharSequence? {
        return content
    }
}